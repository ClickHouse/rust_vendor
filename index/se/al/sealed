{"name":"sealed","vers":"0.6.0","deps":[{"name":"proc-macro2","req":"^1.0","features":[],"optional":false,"default_features":true,"target":null,"kind":null,"package":null},{"name":"quote","req":"^1.0","features":[],"optional":false,"default_features":true,"target":null,"kind":null,"package":null},{"name":"syn","req":"^2.0","features":["extra-traits"],"optional":false,"default_features":true,"target":null,"kind":"dev","package":null},{"name":"syn","req":"^2.0","features":["full"],"optional":false,"default_features":true,"target":null,"kind":null,"package":null},{"name":"trybuild","req":"^1.0","features":["diff"],"optional":false,"default_features":true,"target":null,"kind":"dev","package":null}],"cksum":"22f968c5ea23d555e670b449c1c5e7b2fc399fdaec1d304a17cd48e288abc107","features":{},"yanked":false}