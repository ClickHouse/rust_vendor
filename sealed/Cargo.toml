# THIS FILE IS AUTOMATICALLY GENERATED BY CARGO
#
# When uploading crates to the registry Cargo will automatically
# "normalize" Cargo.toml files for maximal compatibility
# with all versions of Cargo and also rewrite `path` dependencies
# to registry (e.g., crates.io) dependencies.
#
# If you are reading this file be aware that the original Cargo.toml
# will likely look very different (and much more reasonable).
# See Cargo.toml.orig for the original contents.

[package]
edition = "2021"
rust-version = "1.61.0"
name = "sealed"
version = "0.6.0"
authors = ["Jos√© Duarte <jmg.duarte@campus.fct.unl.pt>"]
exclude = ["images/*"]
description = "Macro for sealing traits and structures"
documentation = "https://docs.rs/sealed"
readme = "README.md"
keywords = [
    "proc_macro",
    "sealed",
    "future-proofing",
]
categories = [
    "development-tools",
    "rust-patterns",
]
license = "MIT OR Apache-2.0"
repository = "https://github.com/jmg-duarte/sealed-rs"
resolver = "2"

[lib]
proc-macro = true

[dependencies.proc-macro2]
version = "1.0"

[dependencies.quote]
version = "1.0"

[dependencies.syn]
version = "2.0"
features = ["full"]

[dev-dependencies.syn]
version = "2.0"
features = ["extra-traits"]

[dev-dependencies.trybuild]
version = "1.0"
features = ["diff"]
