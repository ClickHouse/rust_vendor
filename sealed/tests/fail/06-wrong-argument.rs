use sealed::sealed;

#[sealed(erased)]
pub trait T {}

pub struct A;

#[sealed]
impl T for A {}

fn main() {}
