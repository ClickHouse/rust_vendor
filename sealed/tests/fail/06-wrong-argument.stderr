error: unknown `erased` attribute argument
 --> tests/fail/06-wrong-argument.rs:3:10
  |
3 | #[sealed(erased)]
  |          ^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `__seal_t`
 --> tests/fail/06-wrong-argument.rs:8:1
  |
8 | #[sealed]
  | ^^^^^^^^^ use of undeclared crate or module `__seal_t`
  |
  = note: this error originates in the attribute macro `sealed` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0405]: cannot find trait `T` in this scope
 --> tests/fail/06-wrong-argument.rs:9:6
  |
9 | impl T for A {}
  |      ^ not found in this scope
